### ISSUE TITLE
Question: Is it possible to use chrome extensions?

### ISSUE THREAD

**Issue by ranranking** [The-Compiler/qutebrowser#1425]
Since the browser is based on chromium core, is I am correct, is it possible to use chrome extension for now?

**Comment by neeasade** [comment]
it is not based on 'chromium core' currently -- it will be after https://github.com/The-Compiler/qutebrowser/issues/666 , For current plugin plans, ref https://github.com/The-Compiler/qutebrowser/issues/30.

**Comment by ranranking** [comment]
@neeasade I guess I will wait until it is based on chromium core then. Thank you for the reply! So when the chromium implementation is done, it is possible to use current chrome extensions, am I correct?

**Comment by neeasade** [comment]
well, qutebrowser would be moving to be based on qtwebengine, which has a ticket for chrome extension api here: https://trello.com/c/5ZzCOH2j/254-support-some-of-chrome-s-extensions-api

tl;dr maybe

**Comment by ranranking** [comment]
Great. I am using some VPN extensions, which are essential to my work. Hope they will be supported.

**Comment by The-Compiler** [comment]
[QTBUG-51185](https://bugreports.qt.io/browse/QTBUG-51185) has some updated information about this. To be honest, I don't see enough of the Chromium extension API being supported by Qt anytime soon so plugins would work.

As explained in #30 I'll look into Python plugins in the forseeable future, and then at some point I'll also look into supporting JS plugins, maybe with a subset of Chromium's API. But I really can't promise anything currently. For simple extensions, it might be easier to port them to a qutebrowser Python plugin as soon as that exists.

**Comment by jcguu95** [comment]
What's the current state of this issue? The day qb supports chrome (or even Firefox) extensions is the day when qt gets supported by so many people out there!

**Comment by The-Compiler** [comment]
@jcguu95 Nothing new really - as long as QtWebEngine doesn't support WebExtensions it's very unlikely that qutebrowser ever can/will.

**Comment by wardinh0** [comment]
Any news on this issue? I really want to use qutebrowser but there are extensions that I must use

**Comment by The-Compiler** [comment]
Nope.

**Comment by The-Compiler** [comment]
A slightly more verbose answer now that I'm back on a real keyboard: Nothing about this is going to change until (if at all) QtWebEngine supports WebExtensions.

What qutebrowser will offer (and to some degree already does, via Greasemonkey scripts , [userscripts](https://github.com/qutebrowser/qutebrowser/blob/master/doc/userscripts.asciidoc) and its config system) is alternative ways to achieve the same goals as *some* extensions - also see #30. @wardinh0 if you elaborate on what extensions you need exactly, there might be an alternative available.

**Comment by wardinh0** [comment]
@The-Compiler
Thanks for the quick response, Florian! I didn't bother including the extensions I need because I'd feel like demanding it from the devs and it sounded selfish to me. But anyways, I need:
- [Downloader for Instagram™ + Direct Message](https://chrome.google.com/webstore/detail/downloader-for-instagram/olkpikmlhoaojbbmmpejnimiglejmboe)
- [Volume Manager](https://chrome.google.com/webstore/detail/volume-manager/abnhonfioiokelhdappjknfaannlncac)
- [Google Translate](https://chrome.google.com/webstore/detail/google-translate/aapbdbdomjkkjkaonfhkkikfgjllcleb)
- [Private Bookmarks](https://chrome.google.com/webstore/detail/private-bookmarks/cbheenbihjdgbmbogbefcgnpfoflhnhl)
- Any ad block plugin

**Comment by The-Compiler** [comment]
- Downloader for Instagram: You might be able to find a [Greasemonkey userscript](https://greasyfork.org/en/scripts?q=instagram) which works.
- Volume Manager: Not possible as far as I know, though you can mute a tab in qutebrowser (`:tab-mute`)
- Google Translate: Should be possible via an [userscript](https://github.com/qutebrowser/qutebrowser/blob/master/doc/userscripts.asciidoc) (or perhaps a greasemonkey script as well) 
- Private bookmarks: Should also be possible via an userscript (or an external application)
- Adblocking: qutebrowser has a basic host-based adblocker built in, and there's #5317.

**Comment by RomjanHossain** [comment]
sorry to bother you @The-Compiler , but how do we enable those [userscript](https://greasyfork.org/en/scripts?q=instagram)  ? i've paste one of those `script.js` file to `~/.local/share/qutebrowser/userscripts/script.js` file and made it exicutable by `chmod +x script.js`. now how do i make it work in qutebrowser?

**Comment by toofar** [comment]
@RomjanHossain the scripts on greasyfork are greasemonkey scripts (or mostly greasemonkey compatible scripts), they are sometimes called userscripts in general but regarding qutebrowser userscripts are a custom thing to execute a binary file.
Anyway, you have to create a `greasemonkey` folder and put them in there instead of the userscripts folder, see point 18 on https://qutebrowser.org/doc/faq.html

**Comment by emanuele6** [comment]
@RomjanHossain: Also note that greasemonkey scripts don't need to be executable.

**Comment by RomjanHossain** [comment]
btw, is there a way to install firefox_adons/chrome_extension to qutebrowser? I'm curious.
Greasemonkey has limited scripts and most of them are "youtube downloader and Instagram downloader".

**Comment by The-Compiler** [comment]
@RomjanHossain maybe try reading the first couple of comments in this issue? 😉

**Comment by Coder-Takshil** [comment]
> sorry to bother you @The-Compiler , but how do we enable those [userscript](https://greasyfork.org/en/scripts?q=instagram) ? i've paste one of those `script.js` file to `~/.local/share/qutebrowser/userscripts/script.js` file and made it exicutable by `chmod +x script.js`. now how do i make it work in qutebrowser?

then you just have to type ':greasemonkey-reload' which will reload it for you

**Comment by moktavizen** [comment]
> A slightly more verbose answer now that I'm back on a real keyboard: Nothing about this is going to change until (if at all) QtWebEngine supports WebExtensions.

It landed in Qt 6.10

https://doc.qt.io/qt-6/whatsnew610.html#:~:text=Qt%20WebEngine%20Module,QWebEngineExtensionManager%20and%20QWebEngineExtensionInfo
